src/_tutorials/server/cmdline.md

```diff
@@ -73,156 +82,84 @@ The Dart VM supports many options.
+The [args]({{site.pub}}/packages/args) package provides
+parser support for transforming command-line arguments
 into a set of options, flags, and additional values.
 Import the library as follows:
 
-{% prettify dart %}
+<?code-excerpt "misc/bin/dcat.dart" retain="package:args"?>
+```dart
 import 'package:args/args.dart';
-{% endprettify %}
+```
 
-The args library contains two classes:
+The `args` package contains these classes, among others:
 
-| Library | Description |
+| Class | Description |
 |---|---|
-| <a href="{{site.pub-api}}/args/latest/args/ArgParser-class.html" target="_blank">ArgParser</a> | A class that parses command-line arguments |
-| <a href="{{site.pub-api}}/args/latest/args/ArgResults-class.html" target="_blank">ArgResults</a> | The result of parsing command-line arguments using ArgParser. |
+| [ArgParser]({{argsAPI}}/ArgParser-class.html){:target="_blank"} | A command-line argument parser. |
+| [ArgResults]({{argsAPI}}/ArgResults-class.html){:target="_blank"} | The result of parsing command-line arguments using `ArgParser`. |
 {: .table }
 
-Let's take a look at the `dcat` sample,
-which uses ArgParser and ArgResults to parse and store its command-line arguments.
+Here is the `dcat` code that uses these classes to parse and store command-line
+arguments:
 
-<ol>
-<li markdown="1">
-Copy the sample file from the github repo:
-<a href="https://raw.githubusercontent.com/dart-lang/dart-tutorials-samples/master/cmdline/bin/dcat.dart">dcat.dart.</a>
-</li>
+<?code-excerpt "misc/bin/dcat.dart (arg processing)" plaster="none" replace="/(ArgR.*|List[^\)]*|\..*|parser.*|argResults\S[^);]+)/[!$&!]/g"?>
+{% prettify dart %}
+[!ArgResults argResults;!]
 
-<li markdown="1">
-Run the program from the command line as shown by the **boldface** text.
+void main([!List<String> arguments!]) {
+  exitCode = 0; // presume success
+  final [!parser = ArgParser()!]
+    [!..addFlag(lineNumber, negatable: false, abbr: 'n');!]
 
-<pre>
-$ <b>dart dcat.dart -n quotes.txt</b>
-1 Be yourself. Everyone else is taken. -Oscar Wilde
-2 Don't cry because it's over, smile because it happened. -Dr. Seuss
-3 You only live once, but if you do it right, once is enough. -Mae West
-...
-</pre>
 
-The program displays the contents of the source code file and
-preceeds each line with a line number.
-</li>
+  argResults = [!parser.parse(arguments);!]
+  final paths = [!argResults.rest!];
 
-</ol>
+  dcat(paths, [!argResults[lineNumber] as bool!]);
+}
+{% endprettify %}
+
+The runtime passes command-line arguments to the app's `main()` function as a
+list of strings. The `ArgParser` is configured to parse the `-n` flag. The
+result of parsing command-line arguments is stored in `argResults`.
 
 The following diagram shows how the `dcat` command line used above
-is parsed into the `ArgResults` object.
+is parsed into an `ArgResults` object.
 
 ![ArgsParser parses command-line arguments](images/commandlineargs.png)
 
```

Comment:
I agree with your suggested changes. Done. I also changed the named file to "quote.txt" (singular just because I couldn't get "quotes.txt" to fit w/o actually using a Graffle editor to adjust the diagram). I've switched to using "quote.txt" (singular) elsewhere in the page. If you'd prefer to have it plural, then I'll let you do it. Thanks.

Reply:
Nope, the new figure is great.